### DISCUSSION TITLE
Using defaultValues containing null causes an error

### DISCUSSION THREAD

**Discussion by bubblekovrik** [react-hook-form/react-hook-form#4091] [Q&A]
**Problem**
Setting defaultValues which contain some null values and a Controller component produces an error in the console:

> Warning: `value` prop on `input` should not be null. Consider using an empty string to clear the component or `undefined` for uncontrolled components

The null values are received from the api response.

**Codesandbox link**
https://codesandbox.io/s/suspicious-mahavira-zvlvo

**Screenshots**
![image](https://user-images.githubusercontent.com/18058706/106606048-b3abf400-6572-11eb-9dc1-37f12f5e9544.png)

I'm using custom input controls that prevent the rendering when receiving a ``null`` value. Are there ways to avoid setting the values to null from the defaultValues? Setting the ``render`` property instead of ``as`` in the Controller component may solve the problem, but it will requires manual configuration of the input component.

**Comment by bluebill1049** [comment]
```diff
- return <Controller {...props} defaultValue="" as="input" />;
+ return <Controller {...props} as="input" />;
```

you don't need defaultValue there, because you already defined defaultValues at useForm.

**Reply by bubblekovrik** [reply]
Thanks for the quick response! Unfortunately, the original problem still remains. The second error is replaced by a warning from useForm, but the null value is still passed to the input

![image](https://user-images.githubusercontent.com/18058706/106759120-68154b00-6643-11eb-8c78-59fe428ef46d.png)

**Reply by bluebill1049** [reply]
`middleName: undefined` is the reason, change to null or ''

**Reply by bubblekovrik** [reply]
It will remove the warning message, which is annoying but doesn't break anything, but the problem with null values for inputs remains.

**Reply by bluebill1049** [reply]
do a watch or useWatch to toggle that input.

**Reply by bubblekovrik** [reply]
Not sure if I got it right. Should watch/useWatch set the specified defaultValue before the first render?

**Reply by bluebill1049** [reply]
yes at the current version it's safer to provide a default value.

**Comment by leapful** [comment]
@bubblekovrik @bluebill1049 

Another workaround is to use prop "render" of Controller to set default empty string on attribute "value" of input.

Working sample: https://codesandbox.io/s/sparkling-pine-dolzg?file=/src/TextField.js

**Comment by NarenderDhamarlapally** [comment]
first time it is getting null in defaultvalue.
but it is not working as dirty.
firsttime is null .second time is "" empty string .
when i remove value from the textfield
